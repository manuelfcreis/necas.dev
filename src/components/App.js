import React, { Component } from 'react';
import Card from './card.tsx'
import Experience from './experience.tsx'
import Tag from './tags.tsx';
import '../stylesheets/App.css';

class App extends Component {
  render() {
    const items = ['Ruby on Rails', 'Javascript']
    return (
      <div className="App">
        <h1>Manuel Costa Reis</h1>
        <div className='app-body'>
          <div className='left'>
            <Card header='Profile'>
              <p>teste</p>
            </Card>
          </div>

          <div className='right'>
            <Card>
              <h2>Work Experience</h2>
              <Experience dates='2017-current'>
                <h5>Landing.jobs</h5>
                <p>
                  <span className='regular-weight'>Data Scientist & Software Developer</span>
                  <br />
                  Developer etc.
                </p>
                <Tag names={items} />
              </Experience>


              <hr />
              <h2>Education</h2>
              <p></p>
            </Card>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
