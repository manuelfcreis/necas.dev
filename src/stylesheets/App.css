.App {
  width: 100%;
  background-color: #FBFBFB;
}

.App > h1 {
  margin: 1rem;
}

.app-header {
  height: 70px;
  border-bottom: 1px solid #AAA;
}

.app-body {
  display: flex;
}

.left {
  width: 30%;
}

.right {
  width: 70%;
}
