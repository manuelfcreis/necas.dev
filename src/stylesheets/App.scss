@import url('https://fonts.googleapis.com/css?family=Nunito+Sans:400,800');

.app {
  width: 100vw;
  background-color: rgb(148, 12, 12);
  height: 100vh;
  display: flex;
  align-items: center;
  padding: 0;
  font-family: 'Nunito Sans', sans-serif;
  margin: 0;
  position: relative;
}

a, p, h1, h2 {
  font-family: 'Nunito Sans', sans-serif;
}

a {
  color: #ccc;
}

@media screen and (max-width: 1024px) {
  .app {
    flex-direction: column;
    padding: 1rem 0;
    justify-content: center;
  }
}

.container {
  max-height: 90vh;
  overflow-y: scroll;
  margin-top: 4rem 1.5rem 0 0;
  width: 100%;
  display: flex;
  flex-direction: column;
}