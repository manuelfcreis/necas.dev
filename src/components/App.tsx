import React, { Component } from 'react';
import Menu from './menu';
import Eye from './eye';
import Card from './card';
import '../stylesheets/App.scss';
// @ts-ignore
import Div100vh from 'react-div-100vh';


interface iState {
  xPosition: number;
  yPosition: number;
  open: boolean;
}

interface iProps { }

class App extends Component<iProps, iState> {
  constructor(props: iProps) {
    super(props)

    this.state = {
      xPosition: window.innerWidth / 2,
      yPosition: window.innerHeight / 2,
      open: false
    }

    this.menuClick = this.menuClick.bind(this)
  }

  mouseMovement(event: any) {
    event.persist()

    this.setState(() => {
      return ({
        xPosition: event.pageX,
        yPosition: event.pageY
      })
    })
  }

  menuClick() {
    this.setState((prevState) => {
      return ({
        open: !prevState.open
      })
    })
  }

  render() {
    return (
      <Div100vh onClick={this.mouseMovement.bind(this)} onMouseMove={this.mouseMovement.bind(this)} className="app">
        <Menu menuClick={this.menuClick} />
        <Eye xPosition={this.state.xPosition} yPosition={this.state.yPosition} />
        <Card open={this.state.open} />
      </Div100vh>
    );
  }
}

export default App;
