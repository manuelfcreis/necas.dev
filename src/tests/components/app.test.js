import React from "react";
import App from "../../components/App";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const element = shallow(<App />);
  expect(element).toMatchSnapshot();
});
