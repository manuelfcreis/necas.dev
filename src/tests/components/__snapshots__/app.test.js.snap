// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter>
  <Div100vh
    className="app"
    onClick={[Function]}
    onMouseMove={[Function]}
  >
    <withRouter()
      positions={
        Object {
          "x": 512,
          "y": 384,
        }
      }
    />
    <Switch>
      <Route
        component={[Function]}
        path="/blog"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        path="/"
      >
        <Homepage
          positions={
            Object {
              "x": 512,
              "y": 384,
            }
          }
          small={false}
        />
      </Route>
    </Switch>
  </Div100vh>
</BrowserRouter>
`;
